glean-core/src/upload/mod.rs

```diff
@@ -1054,4 +1100,39 @@ mod test {
             PingUploadTask::Done
         );
     }
+
+    #[test]
+    fn quota_is_enforced_when_enqueueing_cached_pings() {
+        let (mut glean, tmpdir) = new_glean(None);
+
+        // Register a ping for testing
+        let ping_type = PingType::new("test", true, /* send_if_empty */ true, vec![]);
+        glean.register_ping_type(&ping_type);
+
+        // Submit the ping multiple times
+        let n = 10;
+        for _ in 0..n {
+            glean.submit_ping(&ping_type, None).unwrap();
+        }
+
+        // Create a new upload manager pointing to the same data_path as the glean instance.
+        let upload_manager = PingUploadManager::new(tmpdir.path(), "Rust", true);
+
+        // Enqueue cached pings and set an arbitrarily low quota.
+        upload_manager.enqueue_cached_pings(&glean, 1);
```

Comment:
An empty ping should have a rather stable size. 
Would it make sense to set the quota to just slightly above that so we see that the last ping is kept and others are deleted?

Reply:
And if we keep `deleted_pings_after_quota_hit` we could check that it deleted the right amount.